ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfilesToggleGroup.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { generatePath, useHistory, useParams } from 'react-router-dom';
+import { ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
+
+import { ComplianceProfileScanStats } from 'services/ComplianceResultsStatsService';
+
+type ProfilesToggleGroupProps = {
+    profiles: ComplianceProfileScanStats[];
+    route: string;
+};
+
+function ProfilesToggleGroup({ profiles, route }: ProfilesToggleGroupProps) {
+    const { profileName: profileNameParam } = useParams();
+    const history = useHistory();
+
+    const handleToggleChange = (selectedProfile) => {
+        const path = generatePath(route, { profileName: selectedProfile });
+        history.push(path);
```

Comment:
What are pro and con of a short comment in the 2 component files about invisible cause-and-effect relationship that this element causes a change to `profileName` parameter value?

Reply:
Oh, an alternative, not sure if I recommend it. What if `handleToggleChange` is a callback function prop?
1. Cause-and-effect relationship is visible in calling component.
2. `generatePath` or new `safeGeneratePath` function:
    * Can use `selectedProfile` arg alone in `ProfileChecksPage`
    * Can use `selectedProfile` arg along with `clusterId` in `ClusterDetailsPage`

Reply:
Great ideas. I hadn't considered using a callback but I think that would be a lot more intuitive. I think I'll do that instead.